src/lektorium/repo/local/storage.py

```diff
@@ -0,0 +1,227 @@
+        self.root = pathlib.Path(root).resolve()
+
+    @property
+    def config(self):
+        config = {}
+        if self.__config_path.exists():
+            with self.__config_path.open('rb') as config_file:
+                def iter_sites(config_file):
+                    config_data = yaml.load(config_file, Loader=yaml.Loader)
+                    for site_id, props in config_data.items():
+                        url = props.pop('url', None)
+                        config = self.site_config(site_id)
+                        name = config.get('project.name')
+                        if name is not None:
+                            props['name'] = name
+                        if url is None:
+                            url = config.get('project.url')
+                        props['production_url'] = url
+                        props['site_id'] = site_id
+                        props.setdefault('name', site_id)
+                        yield props
+                sites = (Site(**props) for props in iter_sites(config_file))
+                config = {s['site_id']: s for s in sites}
+        return self.CONFIG_CLASS(self.__config_path, config)
+
+    def create_session(self, site_id, session_id, session_dir):
+        site_root = self._site_dir(site_id)
+        shutil.copytree(site_root, session_dir)
+
+    def create_site(self, lektor, name, owner, site_id):
+        site_root = self._site_dir(site_id)
+        lektor.quickstart(name, owner, site_root)
+        return site_root, {}
+
+    def site_config(self, site_id):
+        site_root = self._site_dir(site_id)
+        config = list(site_root.glob('*.lektorproject'))
+        if config:
+            return inifile.IniFile(config[0])
+        return collections.defaultdict(type(None))
+
+    def _site_dir(self, site_id):
+        return self.root / site_id / 'master'
+
+    @property
+    def __config_path(self):
+        return self.root / self.CONFIG_FILENAME
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}("{str(self.root)}")'
+
+
+class GitConfig(FileConfig):
+    def __setitem__(self, key, value):
+        super().__setitem__(key, value)
+        parent, name = self.path.parent, self.path.name
+        subprocess.check_call(f'git add {name}', shell=True, cwd=parent)
+        subprocess.check_call('git commit -m autosave', shell=True, cwd=parent)
+        subprocess.check_call('git push origin', shell=True, cwd=parent)
+
+
+class GitStorage(FileStorage):
+    CONFIG_CLASS = GitConfig
+
+    def __init__(self, git):
+        self.git = git
+        self.workdir = pathlib.Path(closer(tempfile.TemporaryDirectory()))
+        lektorium_workdir = self.workdir / 'lektorium'
+        lektorium_workdir.mkdir()
+        subprocess.check_call(
+            f'git clone {self.git} .',
+            shell=True,
+            cwd=lektorium_workdir
+        )
+        super().__init__(lektorium_workdir)
+
+    @classmethod
+    def init(cls, path):
```

Comment:
Why is this `classmethod` but not `staticmethod`?

Reply:
The idea was that each variant of storage implementation should have its own init. You are right - at the moment it has no sense.